subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -547,6 +516,108 @@ int nrf_cloud_encode_config_response(struct nrf_cloud_data const *const input,
 	return 0;
 }
 
+void nrf_cloud_log_control(int log_level)
+{
+	LOG_DBG("Setting nRF Cloud log level = %d", log_level);
+}
+
+int nrf_cloud_encode_control_response(struct nrf_cloud_data const *const input,
+				      struct nrf_cloud_data *const output,
+				      bool *const has_control)
+{
+	__ASSERT_NO_MSG(output != NULL);
+	__ASSERT_NO_MSG(input != NULL);
+
+	char *buffer = NULL;
+	cJSON *root_obj = NULL;
+	cJSON *desired_obj = NULL;
+	cJSON *reported_obj = NULL;
+	cJSON *state_obj = NULL;
+	cJSON *control_obj = NULL;
+	cJSON *alert_obj = NULL;
+	cJSON *log_obj = NULL;
+	cJSON *input_obj = input ? cJSON_Parse(input->ptr) : NULL;
+
+	if (input_obj == NULL) {
+		return -ESRCH; /* invalid input or no JSON parsed */
+	}
+
+	LOG_DBG("Got delta: %s", (const char *)input->ptr);
+	/* A delta update will have the config inside of state */
+	state_obj = cJSON_DetachItemFromObject(input_obj, JSON_KEY_STATE);
+	control_obj = cJSON_DetachItemFromObject(
+		state_obj ? state_obj : input_obj, JSON_KEY_CTRL);
+	if (control_obj) {
+		LOG_DBG("Shadow delta has control section");
+		alert_obj = cJSON_GetObjectItem(control_obj, JSON_KEY_ALERT);
+		if (cJSON_IsBool(alert_obj)) {
+			nrf_cloud_alert_control(cJSON_IsTrue(alert_obj));
+		} else if (alert_obj != NULL) {
+			LOG_DBG("alertsEn is not a bool");
+		} else {
+			LOG_DBG("alertsEn not found");
+		}
+		log_obj = cJSON_GetObjectItem(control_obj, JSON_KEY_LOG);
+		if (cJSON_IsNumber(log_obj)) {
+			nrf_cloud_log_control((int)cJSON_GetNumberValue(log_obj));
+		}
+	} else {
+		LOG_DBG("Shadow delta does not have control section");
+	}
+	cJSON_Delete(input_obj);
+
+	if (has_control) {
+		*has_control = (control_obj != NULL);
+	}
+
+	/* If this is not a delta update, no response data is required */
+	if ((state_obj == NULL) || (control_obj == NULL)) {
+		cJSON_Delete(state_obj);
+		cJSON_Delete(control_obj);
+		output->ptr = NULL;
+		output->len = 0;
+		return 0;
+	}
+
+	/* Prepare JSON response for the delta */
+	root_obj = cJSON_CreateObject();
+	desired_obj = cJSON_AddObjectToObjectCS(root_obj, JSON_KEY_DES);
+	reported_obj = cJSON_AddObjectToObjectCS(root_obj, JSON_KEY_REP);
+
+	/* Add a null control to desired and add the delta control to reported */
+	if (json_add_null_cs(desired_obj, JSON_KEY_CTRL) ||
```

Comment:
i dont _think_ the device should be clearing the desired section...  this was a mistake made with the config implementation.
the cloud should be responsible for setting/changing it.

Reply:
It turns out, if I remove this setting the desired section to null, when I send the update back to the cloud, I get yet another delta notification from the cloud.
If I add setting desired to null back, I do not get a second delta back, which means less traffic.